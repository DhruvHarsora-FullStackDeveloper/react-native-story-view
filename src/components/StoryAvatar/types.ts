import type {
  ImageProps,
  ImageStyle,
  PressableProps,
  TextProps,
  TextStyle,
  ViewStyle,
} from 'react-native';
import type { StoriesType } from '../StoryView/types';

export interface StoryAvatarProps extends StoryAvatarStyleProps {
  item?: StoriesType;
  index: number;
  viewedStories?: Array<boolean[]>;
  openStories?: (position: number) => void;
}

export interface StoryAvatarStyleProps {
  userNameStyle?: TextStyle;
  userImageStyle?: ImageStyle;
  containerStyle?: ViewStyle;
  userImageProps?: ImageProps;
  userNameProps?: TextProps;
  rootProps?: PressableProps;
  viewedStoryContainerStyle?: ViewStyle;
}
