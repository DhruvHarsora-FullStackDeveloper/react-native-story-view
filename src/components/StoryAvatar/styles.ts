import { StyleSheet } from 'react-native';
import { Colors, moderateScale } from '../../theme';

const styles = StyleSheet.create({
  image: {
    height: '95%',
    width: '95%',
    borderRadius: moderateScale(35),
  },
  imageContainer: {
    height: moderateScale(70),
    width: moderateScale(70),
    overflow: 'hidden',
    alignItems: 'center',
    justifyContent: 'center',
    borderRadius: moderateScale(35),
    borderColor: Colors.pink,
    borderWidth: 2,
    marginRight: 10,
  },
});

export default styles;
