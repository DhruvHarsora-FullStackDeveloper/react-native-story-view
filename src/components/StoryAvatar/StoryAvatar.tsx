import React from 'react';
import { View, Pressable, Image } from 'react-native';
import styles from './styles';
import type { StoryAvatarProps } from './types';

const StoryAvatar = ({ item, index, openStories }: StoryAvatarProps) => (
  <Pressable onPress={() => openStories?.(index!)}>
    <View style={styles.imageContainer}>
      <Image
        resizeMode="contain"
        source={{ uri: item?.profile }}
        style={styles.image}
      />
    </View>
  </Pressable>
);
export default StoryAvatar;
