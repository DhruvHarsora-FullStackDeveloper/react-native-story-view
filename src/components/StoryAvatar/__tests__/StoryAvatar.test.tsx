import React from 'react';
import { render } from '@testing-library/react-native';
import StoryAvatar from '../StoryAvatar';

jest.useFakeTimers();

describe('StoryAvatar Component', () => {
  it('Match Snapshot', () => {
    const { toJSON } = render(<StoryAvatar />);
    expect(toJSON()).toMatchSnapshot();
  });
});
