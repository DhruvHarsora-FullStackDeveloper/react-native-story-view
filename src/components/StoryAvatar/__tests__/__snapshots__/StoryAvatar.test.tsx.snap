// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoryAvatar Component Match Snapshot 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderColor": "#CA1D7E",
        "borderRadius": 52.5,
        "borderWidth": 2,
        "height": 105,
        "justifyContent": "center",
        "marginRight": 10,
        "padding": 1,
        "width": 105,
      }
    }
  >
    <Image
      resizeMode="contain"
      source={
        Object {
          "uri": undefined,
        }
      }
      style={
        Object {
          "height": "100%",
          "width": "100%",
        }
      }
    />
  </View>
</View>
`;
